### ISSUE TITLE
Error Building CPLEX.jl

### ISSUE THREAD

**Issue by husseintarhini** [JuliaOpt/CPLEX.jl#257]
I just updated to Julia 1.2 and i have CPLEX 12.8 installed. When building CPLEX.jl i get the following:

(v1.2) pkg> build CPLEX
  Building CPLEX → `C:\Users\husse\.juliapro\JuliaPro_v1.2.0-1\packages\CPLEX\FUtHu\deps\build.log`
┌ Error: Error building `CPLEX`: 
│ ERROR: LoadError: UndefVarError: is_apple not defined
│ Stacktrace:
│  [1] top-level scope at C:\Users\husse\.juliapro\JuliaPro_v1.2.0-1\packages\CPLEX\FUtHu\deps\build.jl:18
│  [2] include at .\boot.jl:328 [inlined]
│  [3] include_relative(::Module, ::String) at .\loading.jl:1094
│  [4] include(::Module, ::String) at .\Base.jl:31
│  [5] include(::String) at .\client.jl:431
│  [6] top-level scope at none:5
│ in expression starting at C:\Users\husse\.juliapro\JuliaPro_v1.2.0-1\packages\CPLEX\FUtHu\deps\build.jl:18
└ @ Pkg.Operations C:\Users\julia\AppData\Local\Julia-1.2.0\share\julia\stdlib\v1.2\Pkg\src\Operations.jl:1138

**Comment by blegat** [comment]
See https://github.com/JuliaOpt/CPLEX.jl/issues/253

**Comment by hei06j** [comment]
I am using Julia 1.1.1 and still get the same error

```
(v1.1) pkg> status
    Status `C:\Users\hei06j\.julia\environments\v1.1\Project.toml`
  [c52e3926] Atom v0.11.2
  [a076750e] CPLEX v0.3.3
  [9961bab8] Cbc v0.6.4
  [b6b21f68] Ipopt v0.6.0
  [4076af6c] JuMP v0.20.0
  [e5e0dc1b] Juno v0.7.2
  [6405355b] Mosek v1.0.4
  [1ec41992] MosekTools v0.9.1
  [91a5bcdd] Plots v0.27.0
  [c36e90e8] PowerModels v0.13.0
  [d7431456] PowerModelsDistribution v0.6.0 [`C:\Users\hei06j\.julia\dev\PowerModelsDistribution`]
```

**Comment by blegat** [comment]
This should work:
```julia
(v1.2) pkg> add CPLEX#master
(v1.2) pkg> build CPLEX
```

**Comment by hei06j** [comment]
Unfortunately go this error:
```
(v1.1) pkg> add CPLEX#master
  Updating registry at `C:\Users\hei06j\.julia\registries\General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Cloning git-repo `https://github.com/JuliaOpt/CPLEX.jl.git`
  Updating git-repo `https://github.com/JuliaOpt/CPLEX.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package LinQuadOptInterface [f8899e07]:
 LinQuadOptInterface [f8899e07] log:
 ├─possible versions are: [0.3.0-0.3.1, 0.4.0-0.4.3, 0.5.0, 0.6.0] or uninstalled
 ├─restricted to versions 0.6.0-0.6 by CPLEX [a076750e], leaving only versions 0.6.0
 │ └─CPLEX [a076750e] log:
 │   ├─possible versions are: 0.5.1 or uninstalled
 │   └─CPLEX [a076750e] is fixed to version 0.5.1
 └─restricted by compatibility requirements with MathOptInterface [b8f27783] to versions: uninstalled — no versions left
   └─MathOptInterface [b8f27783] log:
     ├─possible versions are: [0.5.0-0.5.1, 0.6.0-0.6.4, 0.7.0, 0.8.0-0.8.4, 0.9.0-0.9.5] or uninstalled
     ├─restricted to versions [0.8.0-0.8, 0.9.0-0.9] by PowerModelsDistribution [d7431456], leaving only versions [0.8.0-0.8.4, 0.9.0-0.9.5]
     │ └─PowerModelsDistribution [d7431456] log:
     │   ├─possible versions are: 0.6.0 or uninstalled
     │   └─PowerModelsDistribution [d7431456] is fixed to version 0.6.0
     └─restricted by compatibility requirements with Cbc [9961bab8] to versions: 0.9.1-0.9.5
       └─Cbc [9961bab8] log:
         ├─possible versions are: [0.4.0-0.4.4, 0.5.0-0.5.4, 0.6.0-0.6.4] or uninstalled
         └─restricted to versions 0.6.4 by an explicit requirement, leaving only versions 0.6.4
```

**Comment by husseintarhini** [comment]
I'm getting the following 
`(v1.2) pkg> add CPLEX#master
  Updating git-repo `https://github.com/JuliaOpt/CPLEX.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package LinQuadOptInterface [f8899e07]:
 LinQuadOptInterface [f8899e07] log:
 ├─possible versions are: [0.1.0, 0.2.0, 0.3.0-0.3.1, 0.4.0-0.4.3, 0.5.0, 0.6.0] or uninstalled
 ├─restricted to versions 0.6.0-0.6 by CPLEX [a076750e], leaving only versions 0.6.0
 │ └─CPLEX [a076750e] log:
 │   ├─possible versions are: 0.5.1 or uninstalled
 │   └─CPLEX [a076750e] is fixed to version 0.5.1
 └─restricted by compatibility requirements with MathOptInterface [b8f27783] to versions: uninstalled — no versions left
   └─MathOptInterface [b8f27783] log:
     ├─possible versions are: [0.1.0, 0.2.0, 0.3.0, 0.4.0-0.4.1, 0.5.0-0.5.1, 0.6.0-0.6.4, 0.7.0, 0.8.0-0.8.4, 0.9.0-0.9.5] or uninstalled
     └─restricted to versions 0.9.5 by an explicit requirement, leaving only versions 0.9.5`

**Comment by blegat** [comment]
@hei06j Cbc v0.6.4 needs MOI v0.9 and CPLEX needs MOI v0.8 (not yet updated to MOI v0.9). Remove Cbc or don't require that it's using exactly the version v0.6.4 to solve the issue.

@husseintarhini You explicitly require that MOI is v0.9.5 but MOI only supports MOI v0.8 yet.

Note that MOI means MathOptInterface

**Comment by hei06j** [comment]
Thanks @blegat, but still didn't work. I am not sure, but I think MOI v9.0 is being used by a few other packages. Please see the following errors after removing Cbc
```
(v1.1) pkg> add CPLEX#master
  Updating git-repo `https://github.com/JuliaOpt/CPLEX.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package LinQuadOptInterface [f8899e07]:
 LinQuadOptInterface [f8899e07] log:
 ├─possible versions are: [0.3.0-0.3.1, 0.4.0-0.4.3, 0.5.0, 0.6.0] or uninstalled
 ├─restricted to versions 0.6.0-0.6 by CPLEX [a076750e], leaving only versions 0.6.0
 │ └─CPLEX [a076750e] log:
 │   ├─possible versions are: 0.5.1 or uninstalled
 │   └─CPLEX [a076750e] is fixed to version 0.5.1
 └─restricted by compatibility requirements with MathOptInterface [b8f27783] to versions: uninstalled — no versions left
   └─MathOptInterface [b8f27783] log:
     ├─possible versions are: [0.5.0-0.5.1, 0.6.0-0.6.4, 0.7.0, 0.8.0-0.8.4, 0.9.0-0.9.5] or uninstalled
     ├─restricted to versions [0.8.0-0.8, 0.9.0-0.9] by PowerModelsDistribution [d7431456], leaving only versions [0.8.0-0.8.4, 0.9.0-0.9.5]
     │ └─PowerModelsDistribution [d7431456] log:
     │   ├─possible versions are: 0.6.0 or uninstalled
     │   └─PowerModelsDistribution [d7431456] is fixed to version 0.6.0
     ├─restricted by compatibility requirements with Ipopt [b6b21f68] to versions: 0.9.0-0.9.5
     │ └─Ipopt [b6b21f68] log:
     │   ├─possible versions are: [0.4.1-0.4.5, 0.5.0-0.5.4, 0.6.0] or uninstalled
     │   └─restricted to versions 0.6.0 by an explicit requirement, leaving only versions 0.6.0
     └─restricted by compatibility requirements with MosekTools [1ec41992] to versions: 0.9.5
       └─MosekTools [1ec41992] log:
         ├─possible versions are: [0.6.0, 0.7.0-0.7.4, 0.8.0, 0.9.0-0.9.1] or uninstalled
         └─restricted to versions 0.9.1 by an explicit requirement, leaving only versions 0.9.1

(v1.1) pkg> status
    Status `C:\Users\hei06j\.julia\environments\v1.1\Project.toml`  [c52e3926] Atom v0.11.2
  [b6b21f68] Ipopt v0.6.0
  [4076af6c] JuMP v0.20.0
  [e5e0dc1b] Juno v0.7.2
  [6405355b] Mosek v1.0.4
  [1ec41992] MosekTools v0.9.1
  [91a5bcdd] Plots v0.27.0
  [c36e90e8] PowerModels v0.13.0
  [d7431456] PowerModelsDistribution v0.6.0 [`C:\Users\hei06j\.julia\dev\PowerModelsDistribution`]
```

**Comment by hei06j** [comment]
My issue is fixed, I had to remove everything and add CPLEX first, then JuMP and the rest of the packages. Thanks again.

**Comment by blegat** [comment]
Yes, Ipopt and MosekTools are requiring MOI v0.9 it seems. As @hei06j says, the easiest is to remove Ipopt and MosekTools and add CPLEX or MOI v0.8 or JuMP v0.19 first.

**Comment by husseintarhini** [comment]
Thanks @hei06j , removing all the packages then adding CPLEX first worked.

**Comment by odow** [comment]
Closed by the v0.6.0 tag: https://github.com/JuliaOpt/CPLEX.jl/releases/tag/v0.6.0.
